x/evmutil/keeper/conversion_evm_native.go

```diff
@@ -71,7 +71,15 @@ func (k Keeper) ConvertCoinToERC20(
 		return err
 	}
 
-	if err := k.UnlockERC20Tokens(ctx, pair, coin.Amount.BigInt(), receiverAccount); err != nil {
+	// handle bep3 conversion pair (8 decimals sdk.Coin -> 18 decimals erc20)
+	// when converting bep3 bank coins to erc20, we will need to unlock the 18
+	// decimals erc20 equivalent of the 8 decimals bep3 sdk.Coins
+	amountToUnlock := coin.Amount.BigInt()
+	if IsEvmNativeBep3Conversion(pair) {
```

Comment:
After a map is used & conversion factor is pulled to a top level variable, I wonder if 3 private methods are necessary.

option 1
```golang
// coin -> erc20
amountToUnlock := coin.Amount
if _, isBep3Asset := bep3Denoms[coin.Denom]; isBep3Asset {
  amountToUnlock = amountToUnlock.Mul(bep3ConversionFactor)
}

// erc20 -> coin
amountToLock := amount
amountToMint := amount
if _, isBep3Asset := bep3Denoms[pair.Denom]; isBep3Asset {
  amountToMint = amountToMint.Quo(bep3ConversionFactor)

  if amountToMint.IsZero() {
    return errorsmod.Wrapf(types.ErrInsufficientConversionAmount, "unable to convert bep3 coin due converting less than 1%s", pair.Denom)
  }

  amountToLock := amountToMint.Mul(bep3ConversionFactor)
}
```

vs option 2

```golang
// coin -> erc20
amountToUnlock := coin.Amount
if isBep3Asset(coin.Denom) {
  amountToUnlock = covertBep3CoinToErc20(amountToUnlock)
}

// erc20 -> coin
amountToLock := amount
amountToMint := amount
if isBep3Asset(pair.Denom) {
  amountToMint := convertBep3Erc20ToCoin(amountToMint)

  if amountToMint.IsZero() {
    return errorsmod.Wrapf(types.ErrInsufficientConversionAmount, "unable to convert bep3 coin due converting less than 1%s", pair.Denom)
  }

  amountToLock := convertBep3CoinToErc20(amountToLock)
```

vs option 3

```golang
// coin -> erc20
amountToUnlock := convertToErc20Decimals(coin)

// erc20 -> coin
amountToLock, amountToMint, err := calculateAmountToLockAndMint(pair.Denom, amount)
if err != nil {
  return err
}
```

- Option 1 is the default implementation, but doesn't match the complexity level of the calling function well.
- Option 2 it may use some named functions, but is the complexity of the top level function actually reduced?  Also, is `amountToLock := convertBep3CoinToErc20(amountToLock)` actually more comprehensible than `amountToLock := amountToMint.Mul(bep3ConversionFactor)`?  Enough so that the extra function is worth it?
- Option 3 naming could be better, but this pushes the complexity down a level -- improving readability at the top level, but also keeping the implementation easy to find & understand.  IMO, this seems like the best balance.

Reply:
I do prefer option 2's separation of the bep3 logic with the rest of the conversion logic. It's very clear when reading the conversion functions that bep3 assets are handled differently, and we can easily remove that logic in the future without touching any existing conversion logic. 
I do also think abstracting some of the bep3 logic will help with readability, especially for the erc20 -> coin conversion logic. I'll test out a hybrid of option 2 and 3's approach.

Reply:
Cool 👍. Ya do agree option 3 doesn't clearly show it's for bep3 only, hybrid approach would be good

Reply:
https://github.com/Kava-Labs/kava/pull/1848/commits/98fc5962aaee656d87004abbf367211b1db70eca